### ISSUE TITLE
Upgrade To CloudEvents V2 (& Tracing)

### ISSUE THREAD

**Issue by travis-minke-sap** [kyma-incubator/knative-kafka#108]
Currently the implementation is based on CloudEvents V1 and we need to catch up with the rest of knative eventing by moving to V2.  Also, our current implementation was based on Deprecated / Removed functionality in the knative eventing cloudevent client.  Specifically this was done to facilitate Tracing with Istio/Graphana.  We need to take a step back and determine the appropriate tracing requirements and alignment with knative and implement accordingly. 

Make sure to bump the CloudEvents sdk-go dependency forward to latest V2 version as well!

**Comment by travis-minke-sap** [comment]
After reviewing the current knative implementation the tracing work has been broken out into a separate Issue... #125 and will not be addressed herein.  The propagation of all `ce-xxx` headers is working but the use of W3C Tracecontext headers (`traceparent`) is not.

**Comment by travis-minke-sap** [comment]
Once this is complete we will create a patch to update the [knative-sandbox/eventing-kafka](https://github.com/knative-sandbox/eventing-kafka) project as called for in [Issue #7](https://github.com/knative-sandbox/eventing-kafka/issues/7).

This is expected to be the final set of work performed in this repo (other than releasing v0.14.0) with all future development occurring in the new sandbox repo.

**Comment by travis-minke-sap** [comment]
PR has been merged into both the kyma-incubator and the knative-sandbox.
